### DISCUSSION TITLE
Is there any best practice to avoid module-level singleton issue on next js?

### DISCUSSION THREAD

**Discussion by chiralium** [pmndrs/zustand#2259] [Q&A]
How you dealing with that?

**Comment by dai-shi** [comment]
Use React Context: https://github.com/pmndrs/zustand/blob/a4578fa5c48c013408744eddd7d051ade223d076/docs/guides/initialize-state-with-props.md?plain=1#L80

@dbritto-dev Maybe we should have a specific guide named Next.js or something.

**Reply by chiralium** [reply]
If I'm using no-store actions that calling inside a `getServerSiderProps`-function. How to properly get the store to call action on server side?

**Reply by dbritto-dev** [reply]
@dai-shi we shall create a guide to talk about zustand and how you can use it with nextjs page router and app router

**Reply by chiralium** [reply]
It would be great.

**Reply by ootkin** [reply]
Following for nextjs app router
